android/app/src/main/java/org/musicpd/NetworkUtil.java

```diff
@@ -0,0 +1,31 @@
+package org.musicpd;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.LinkAddress;
+import android.net.LinkProperties;
+
+import java.net.InetAddress;
+import java.util.List;
+
+public class NetworkUtil {
+
+    public static String getDeviceIPV4Address(Context context) {
+        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        LinkProperties linkProperties = connectivityManager.getLinkProperties(connectivityManager.getActiveNetwork());
+        if (linkProperties != null) {
+            List<LinkAddress> linkAddresses = linkProperties.getLinkAddresses();
+            for (LinkAddress address : linkAddresses) {
+                if (!address.getAddress().isLinkLocalAddress() && !address.getAddress().isLoopbackAddress()) {
+                    InetAddress address1 = address.getAddress();
+                    String hostAddress = address1.getHostAddress();
+                    if (hostAddress != null && hostAddress.contains(".")) {
```

Comment:
`instanceof Inet4Address`?

Reply:
Ah yeah that's definitely cleaner, thanks.